### PR TITLE
Added steps for App Registration pre-creation using existing deployment tools.

### PR DESCRIPTION
**Pull request by karok2m**
Changes:
* Added App Registration pre-creation steps using `deploy.ps1` and `aad-register.ps1`.
* Added App Registration pre-creation steps using `Microsoft.Azure.IIoT.Deployment`.
* Removed language references from links in `howto-deploy-all-in-one.md`. Minor formatting changes.

### CODE REVIEW

File: docs/deploy/howto-deploy-all-in-one.md [hunk 1]
```diff
@@ -121,7 +124,7 @@ Start the deployment
 
 On **Windows**, the script uses Powershell, which comes with Windows. The deploy batch file uses it to install all required modules.
 
-In case you run into issues, e.g. because you want to use pscore, run following two commands in PowerShell as Administrator. See more information about [AzureAD and Az modules](https://docs.microsoft.com/en-us/powershell/azure/install-az-ps).
+In case you run into issues, e.g. because you want to use pscore, run following two commands in PowerShell as Administrator. See more information about [AzureAD and Az modules](https://docs.microsoft.com/powershell/azure/install-az-ps).
```
Comment (barnstee) [inline docs/deploy/howto-deploy-all-in-one.md:127]: missing "the"

File: docs/deploy/howto-register-aad-applications.md [hunk 2]
```diff
@@ -1,21 +1,141 @@
...
-* App registration for AKS cluster.
+* [Introduction](#introduction)
+* [PowerShell scripts](#powershell-scripts)
+* [Microsoft.Azure.IIoT.Deployment](#microsoftazureiiotdeployment)
 
-## ServicesApp
+## Introduction
 
-TBD
+Components of Azure Industrial IoT solution require several App Registrations in your Azure Active Directory
```
Comment (barnstee) [inline docs/deploy/howto-register-aad-applications.md:13]: missing "the"

File: docs/deploy/howto-register-aad-applications.md [hunk 3]
```diff
@@ -1,21 +1,141 @@
...
+(AAD) to run. Those App Registrations are responsible for:
 
-## ClientsApp
+* providing identity for back-end microservices to run
+* defining authentication methods and details for client applications
 
-TBD
+Both of our deployment methods (`deploy.ps1` script or `Microsoft.Azure.IIoT.Deployment` application) create
+those App Registrations for you by default. But they both require user to have `Administrator` role in the AAD
```
Comment (barnstee) [inline docs/deploy/howto-register-aad-applications.md:20]: missing "the"

File: docs/deploy/howto-register-aad-applications.md [hunk 4]
```diff
@@ -1,21 +1,141 @@
...
+for deployment to succeed. In cases when the user does not have the `Administrator` role, the process of
+deployment can be separated into 2 distinct states:
 
-## AKS
+* creation and setup of App Registrations in AAD tenant
```
Comment (barnstee) [inline docs/deploy/howto-register-aad-applications.md:24]: missing "the"

File: docs/deploy/howto-register-aad-applications.md [hunk 5]
```diff
@@ -1,21 +1,141 @@
...
+
+## PowerShell scripts
+
+`deploy.ps1` PowerShell script calls `aad-register.ps1` for creating the following App Registrations:
+
+* `<application-name>-service`: used for providing identity for back-end microservices
+* `<application-name>-client`: used for authentication of native clients, such as CLI tool
+* `<application-name>-web`: used for authentication of Web clients, such as Swagger
+
+Use the following command to run `aad-register.ps1` script for creation of App Registrations.
```
Comment (barnstee) [inline docs/deploy/howto-register-aad-applications.md:38]: missing "the"

File: docs/deploy/howto-register-aad-applications.md [hunk 6]
```diff
@@ -1,21 +1,141 @@
...
+Specify desired application name instead of `<application-name>` and your tenant id.
```
Comment (barnstee) [inline docs/deploy/howto-register-aad-applications.md:39]: missing "the"

File: docs/deploy/howto-register-aad-applications.md [hunk 7]
```diff
@@ -1,21 +1,141 @@
...
+Follow script commands and provide additional details where needed.
```
Comment (barnstee) [inline docs/deploy/howto-register-aad-applications.md:40]: missing "the"

File: docs/deploy/howto-register-aad-applications.md [hunk 8]
```diff
@@ -1,21 +1,141 @@
...
+
+```bash
+cd deploy/scripts
+./aad-register.ps1 -Name <application-name> -ReplyUrl https://<application-name>.azurewebsites.net/ -TenantId XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX -Output aad.json
+```
+
+The output of the script is a JSON file containing the relevant information to be used as part of the
+deployment and must be passed to the `deploy.ps1` script in the same folder using `-aadConfig` argument.
+
+Use the following command to deploy Azure Industrial IoT platform with pre-created App Registrations.
```
Comment (barnstee) [inline docs/deploy/howto-register-aad-applications.md:53]: missing "the"

File: docs/deploy/howto-register-aad-applications.md [hunk 9]
```diff
@@ -1,21 +1,141 @@
...
+Follow script commands and provide additional details where needed.
```
Comment (barnstee) [inline docs/deploy/howto-register-aad-applications.md:54]: missing "the"

File: docs/deploy/howto-register-aad-applications.md [hunk 10]
```diff
@@ -1,21 +1,141 @@
...
+
+```bash
+./deploy.ps1 -aadConfig aad.json -applicationName <application-name>
+```
+
+You can find all configuration options for `deploy.ps` [here](./howto-deploy-all-in-one.md#deployment-script-options).
+
+## Microsoft.Azure.IIoT.Deployment
+
+Find links for downloading executable `Microsoft.Azure.IIoT.Deployment` application [here](./howto-deploy-aks.md#download-microsoftazureiiotdeployment-binaries).
```
Comment (barnstee) [inline docs/deploy/howto-register-aad-applications.md:64]: missing "the"

File: docs/deploy/howto-register-aad-applications.md [hunk 11]
```diff
@@ -1,21 +1,141 @@
...
+
+`Microsoft.Azure.IIoT.Deployment` application creates the following App Registrations:
+
+* `<application-name>-service`: used for providing identity for back-end microservices
+* `<application-name>-client`: used for authentication of native and Web clients, such as CLI tool or Swagger
+* `<application-name>-aks`: used for providing identity for AKS cluster
+
+Use the following command to run `Microsoft.Azure.IIoT.Deployment` application for App Registration creation
```
Comment (barnstee) [inline docs/deploy/howto-register-aad-applications.md:72]: missing "the"

File: docs/deploy/howto-register-aad-applications.md [hunk 12]
```diff
@@ -1,21 +1,141 @@
...
+> using a DNS entry for a Public IP Address resource to provide a URL for the Ingress. So please make sure to
+> use correct format for the region that you want to deploy to.
+
+```bash
+.\Microsoft.Azure.IIoT.Deployment.exe RunMode=ApplicationRegistration Auth:AzureEnvironment=AzureGlobalCloud ApplicationName=<application-name> ApplicationUrl=<application-name>.northeurope.cloudapp.azure.com
+```
+
+The command will output a JSON object that contains details and credentials of created App Registrations and
+associated Service Principals. Those should be used as an input for the next step. To do that, please create
+a `appsettings.json` file in the same directory as `Microsoft.Azure.IIoT.Deployment` application and use the
```
Comment (barnstee) [inline docs/deploy/howto-register-aad-applications.md:85]: missing "the"

File: docs/deploy/howto-register-aad-applications.md [hunk 13]
```diff
@@ -1,21 +1,141 @@
...
+JSON object as value of `ApplicationRegistration` key as shown bellow.
```
Comment (barnstee) [inline docs/deploy/howto-register-aad-applications.md:86]: missing "the"

File: docs/deploy/howto-register-aad-applications.md [hunk 14]
```diff
@@ -1,21 +1,141 @@
...
+        "AksApplicationSP": {
+            "Id": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
+            "DisplayName": "<application-name>-aks"
+        },
+        "AksApplicationSecret": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+    }
+}
+```
+
+After that, use the following command to deploy Azure Industrial IoT platform with pre-created App
```
Comment (barnstee) [inline docs/deploy/howto-register-aad-applications.md:134]: missing "the"

File: docs/deploy/howto-register-aad-applications.md [hunk 15]
```diff
@@ -1,21 +1,141 @@
...
+Registrations. Follow the commands and provide additional details where needed.
+
+```bash
+.\Microsoft.Azure.IIoT.Deployment.exe RunMode=ResourceDeployment Auth:AzureEnvironment=AzureGlobalCloud ApplicationName=<application-name> ResourceGroup:Region=EuropeNorth
+```
+
+You can find all configuration options for `Microsoft.Azure.IIoT.Deployment` [here](./howto-deploy-aks.md#configuration).
```
Comment (barnstee) [inline docs/deploy/howto-register-aad-applications.md:141]: missing "the" and missing "application"

**Review by barnstee — CHANGES_REQUESTED** [review]

File: docs/deploy/howto-register-aad-applications.md (hunk 2 above)
Comment (karok2m) [reply docs/deploy/howto-register-aad-applications.md:13]: Fixed.

File: docs/deploy/howto-deploy-all-in-one.md (hunk 1 above)
Comment (karok2m) [reply docs/deploy/howto-deploy-all-in-one.md:127]: the Administrator ?

File: docs/deploy/howto-register-aad-applications.md (hunk 4 above)
Comment (karok2m) [reply docs/deploy/howto-register-aad-applications.md:24]: Fixed.

File: docs/deploy/howto-register-aad-applications.md (hunk 5 above)
Comment (karok2m) [reply docs/deploy/howto-register-aad-applications.md:38]: Fixed.

File: docs/deploy/howto-register-aad-applications.md (hunk 6 above)
Comment (karok2m) [reply docs/deploy/howto-register-aad-applications.md:39]: Fixed.

File: docs/deploy/howto-register-aad-applications.md (hunk 7 above)
Comment (karok2m) [reply docs/deploy/howto-register-aad-applications.md:40]: Fixed.

File: docs/deploy/howto-register-aad-applications.md (hunk 9 above)
Comment (karok2m) [reply docs/deploy/howto-register-aad-applications.md:54]: Fixed.

File: docs/deploy/howto-register-aad-applications.md (hunk 8 above)
Comment (karok2m) [reply docs/deploy/howto-register-aad-applications.md:53]: Fixed.

File: docs/deploy/howto-deploy-all-in-one.md (hunk 1 above)
Comment (barnstee) [reply docs/deploy/howto-deploy-all-in-one.md:127]: run the following

File: docs/deploy/howto-register-aad-applications.md (hunk 10 above)
Comment (karok2m) [reply docs/deploy/howto-register-aad-applications.md:64]: Fixed.

File: docs/deploy/howto-register-aad-applications.md (hunk 11 above)
Comment (karok2m) [reply docs/deploy/howto-register-aad-applications.md:72]: Where ? `the  Microsoft.Azure.IIoT.Deployment application` or `the App Registration` ?

File: docs/deploy/howto-register-aad-applications.md (hunk 12 above)
Comment (karok2m) [reply docs/deploy/howto-register-aad-applications.md:85]: Fixed.

File: docs/deploy/howto-register-aad-applications.md (hunk 13 above)
Comment (karok2m) [reply docs/deploy/howto-register-aad-applications.md:86]: Fixed.

File: docs/deploy/howto-register-aad-applications.md (hunk 11 above)
Comment (barnstee) [reply docs/deploy/howto-register-aad-applications.md:72]: the Microsoft.Azure.IIoT.Deployment application

File: docs/deploy/howto-register-aad-applications.md (hunk 14 above)
Comment (karok2m) [reply docs/deploy/howto-register-aad-applications.md:134]: Fixed.

File: docs/deploy/howto-register-aad-applications.md (hunk 15 above)
Comment (karok2m) [reply docs/deploy/howto-register-aad-applications.md:141]: Fixed.

**Review by barnstee — DISMISSED** [review]

File: docs/deploy/howto-deploy-all-in-one.md (hunk 1 above)
Comment (karok2m) [reply docs/deploy/howto-deploy-all-in-one.md:127]: Fixed.

File: docs/deploy/howto-register-aad-applications.md (hunk 11 above)
Comment (karok2m) [reply docs/deploy/howto-register-aad-applications.md]: Fixed.

File: docs/deploy/howto-register-aad-applications.md (hunk 3 above)
Comment (karok2m) [reply docs/deploy/howto-register-aad-applications.md:20]: the `deploy.ps1` script or the `Microsoft.Azure.IIoT.Deployment` application, right ?

Comment (karok2m) [reply docs/deploy/howto-register-aad-applications.md:20]: Fixed, added the in front of `Microsoft.Azure.IIoT.Deployment` application.

**Review by barnstee — APPROVED** [review]

**Pull request merged**